### DISCUSSION TITLE
Can DB viewer store result history?

### DISCUSSION THREAD

**Discussion by sovereignstack** [dbeaver/dbeaver#10679] [Q&A]
Sometimes I want to see the result of a previously executed query. At present I find no option to do this. So I am forced to execute the query again. Which is bad especially if I have it is an expensive query. I do see Back and Forward buttons but they are greyed and don't have any mouse-hover tooltip, so I don't know what they are for.

I could not find an option to enable result history in settings.

**Comment by serge-rider** [comment]
You can execute different queries in separate tabs (<kbd>ctrl+\ </kbd>). 
Also you can pin tabs to avoid their close by accident.

**Reply by sovereignstack** [reply]
Thanks, but this is not the same as a history. I run many queries so, query per tab does not suit what I need. 

Do you happen to know what the back and forward buttons in the result set pane are? 
![image](https://user-images.githubusercontent.com/3187668/102173012-17456980-3ec0-11eb-88fa-7417384ba05c.png)

**Reply by serge-rider** [reply]
You can see historical queries in the Query Manager view. Or on the Execution Log tab in SQL Editor.

These buttons are for foreign key navigation. If your tables have foreign keys then you will see small link icons and you can navigate by them and then got back and forward.

**Reply by sovereignstack** [reply]
Query history, just stores the query not the result. For the results, you have to execute the query again. This takes time and also unnecessarily hammers the database if the query is expensive. Is there any plan to add this feature? Even the basic Eclipse database perspective has it, so I am really missing that feature in DBeaver.

**Reply by serge-rider** [reply]
Well, executing query in a separate tab + pin that tab usually used for such tasks.

Could you say how is it implmeneted in Eclipse database perspective? Honestly I don't remember such feature there..

**Reply by Xethron** [reply]
Found this discussion while looking for something similar. Not sure if I should open a new thread?

In short, I'd like to see the results of previous queries ran.

An example use case would be: Run a query to find bad data, fix bad data, refresh query to see everything is fixed, but you forgot to save the bad data to debug/investigate on later. Since the data was already deleted/fixed/changed, there is no way to re-run the query in a different tab. A history of results would've been nice. Even if it's just in a log file somewhere.
